packages/validator/src/services/validatorStore.ts

```diff
@@ -191,18 +223,79 @@ export class ValidatorStore {
   }
 
   getGasLimit(pubkeyHex: PubkeyHex): number {
-    return (
-      (this.validators.get(pubkeyHex)?.builder || {}).gasLimit ??
-      this.defaultProposerConfig?.builder.gasLimit ??
-      defaultOptions.defaultGasLimit
-    );
+    if (this.validators.has(pubkeyHex)) {
+      return (
+        (this.validators.get(pubkeyHex)?.builder || {}).gasLimit ??
+        this.defaultProposerConfig?.builder.gasLimit ??
+        defaultOptions.defaultGasLimit
+      );
+    } else {
+      throw Error(`Validator pubkey ${pubkeyHex} not known`);
+    }
+  }
+
+  setGasLimit(pubkeyHex: PubkeyHex, gasLimitInput: string | number): void {
+    if ((typeof gasLimitInput !== "string" && typeof gasLimitInput !== "number") || `${gasLimitInput}`.trim() === "") {
+      throw Error("Not valid Gas Limit");
+    }
+    const gasLimit = Number(gasLimitInput);
+    if (Number.isNaN(gasLimit) || gasLimit === 0) {
+      throw Error(`Gas Limit is not valid gasLimit=${gasLimit}`);
+    }
+
+    if (this.validators.has(pubkeyHex)) {
+      const validatorData = this.validators.get(pubkeyHex);
+      if (validatorData !== undefined) {
+        this.validators.set(pubkeyHex, {...validatorData, builder: {...validatorData.builder, gasLimit}});
+      } else {
+        throw Error(`ValidatorData for pubkey ${pubkeyHex} not set `);
+      }
+    } else {
+      throw Error(`Validator pubkey ${pubkeyHex} not known`);
+    }
+  }
+
+  deleteGasLimit(pubkeyHex: PubkeyHex): void {
+    if (this.validators.has(pubkeyHex)) {
+      const validatorData = this.validators.get(pubkeyHex);
+      if (validatorData !== undefined) {
+        delete validatorData.builder["gasLimit"];
+        this.validators.set(pubkeyHex, validatorData);
+      } else {
+        throw Error(`ValidatorData for pubkey ${pubkeyHex} not set `);
+      }
+    } else {
+      throw Error(`Validator pubkey ${pubkeyHex} not known`);
+    }
   }
 
   /** Return true if `index` is active part of this validator client */
   hasValidatorIndex(index: ValidatorIndex): boolean {
     return this.indicesService.index2pubkey.has(index);
   }
 
+  getProposerConfig(pubkeyHex: PubkeyHex): ProposerConfig | null {
+    let proposerConfig: ProposerConfig | null = null;
+    if (this.validators.has(pubkeyHex)) {
+      const validatorData = this.validators.get(pubkeyHex);
+      if (validatorData !== undefined) {
+        const {graffiti, strictFeeRecipientCheck, feeRecipient, builder} = validatorData;
+
+        // if anything is set , i.e not default then return
+        if (
+          graffiti !== undefined ||
+          strictFeeRecipientCheck !== undefined ||
+          feeRecipient !== undefined ||
+          builder.enabled !== undefined ||
+          builder.gasLimit !== undefined
+        ) {
+          proposerConfig = {graffiti, strictFeeRecipientCheck, feeRecipient, builder};
```

Comment:
Does the ValidatorStore really need to know about the idea of a proposerConfig? Feels like it's the API role to construct this object from the other methods

Reply:
ummm i don't find the idea of ValidatorStore knowing about the proposer config odd as we also pass proposer data in the addSigner now. 
Other getters anyway return defaults if the that particular property is not assigned.